zircon.core/src/commonMain/kotlin/org/hexworks/zircon/api/dsl/component/TextBoxDsl.kt

```diff
@@ -2,22 +2,28 @@ package org.hexworks.zircon.api.dsl.component
 
 import org.hexworks.zircon.api.builder.component.ParagraphBuilder
 import org.hexworks.zircon.api.builder.component.TextBoxBuilder
-import org.hexworks.zircon.api.component.Component
+import org.hexworks.zircon.api.component.Container
+import org.hexworks.zircon.api.component.HBox
 import org.hexworks.zircon.api.component.TextBox
-import org.hexworks.zircon.api.data.Position
+import org.hexworks.zircon.api.component.builder.base.BaseContainerBuilder
 
-fun textBox(
+fun buildTextBox(
     initialContentWidth: Int,
-    nextPosition: Position = Position.defaultPosition(),
-    components: MutableList<Component> = mutableListOf(),
     init: TextBoxBuilder.() -> Unit
-): TextBox =
-    TextBoxBuilder(initialContentWidth, nextPosition, components).apply(init).build()
+): TextBox = TextBoxBuilder(initialContentWidth).apply(init).build()
+
+/**
+ * Creates a new [HBox] using the component builder DSL, adds it to the
+ * receiver [Container] it and returns the [HBox].
+ */
+fun <T : BaseContainerBuilder<*, *>> T.textBox(
+    initialWidth: Int,
+    init: TextBoxBuilder.() -> Unit
+): TextBox {
+    val result = TextBoxBuilder(initialWidth).apply(init).build()
+    this.withAddedChildren(result)
+    return result
+}
 
 fun TextBoxBuilder.paragraph(withNewLine: Boolean = true, init: ParagraphBuilder.() -> Unit) =
     addParagraph(ParagraphBuilder().apply(init).build(), withNewLine)
```

Comment:
I'm thinking this should likely be duplicated for the header?

Reply:
That's what I was thinking too, but it is unnecessary because there is nothing to build (the `TextBox` manages the `Header`'s placement).

Reply:
don't we still want to be able to add a header in a dsl style? because TextBox is not a Container, there is no way to write
```
buildTextBox {
   header {}
   paragraph {}
}
```
without adding `TextBoxBuilder.header()` here I believe

Reply:
found an example where a header is made
```
textBox(column.contentWidth - 5) {
                    name = "Text Box"
                    decorations = box() + shadow()

                    addHeader("Text Box!")

                    paragraph {
                        +"Paragraph text."
                    }
                }
```
and now better understand what you mean. What I'm thinking is, a HeaderBuilder can have decoration set to it for example. Is this something we're explicitly disallowing for headers inside a TextBox? if so, i think this is a perfectly acceptable way of doing that over creating a separate TextBoxHeaderBuilder for the `TextBoxBuilder.header {}` scope

Reply:
The problem is that right now the `addHeader` function accepts a `String`, not a `Header` or a `HeaderBuilder`. I'm going to deprecate this and add a new version that uses the builder.

Reply:
I also added a builder for `ListItem` which was missing altogether.